### ISSUE TITLE
Is Finagle still maintained?

### ISSUE THREAD

**Issue by kiesia-sc** [twitter/finagle#943]
Hi folks 👋 is this project still maintained? I see some housekeeping commits, but will you be actively developing new features?

**Comment by xin301x** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by shakeelrao** [comment]
I don't believe so. All of the core contributors are no longer employed by Twitter.

**Comment by felixbr** [comment]
I think the answer at this point is [no](https://www.zdnet.com/article/twitter-turns-its-back-on-open-source-development/), at least not publicly on github.

The bigger question is if we should try to fork and maintain the open-source version of Finagle separately going forward. By "we" I mean companies and people that still use Finagle for production deployments and have no immediate interest in replacing it with a different solution anytime soon.

Finagle is still a very robust framework and I haven't seen many alternatives that have the same features or reliability.

I would offer some of my time to keep basic maintenance up (like [Scala versions](https://github.com/twitter/util/issues/274), newer [JDK versions](https://endoflife.date/java), security updates). But I cannot do everything alone as this is a big project and a bus factor of 1 is always bad. My hope would be that devs from other companies which use Finagle will share the responsibility in a collaboration effort.

I'm tagging some people I've had the pleasure to collaborate with on Finagle and related projects in the past who might also still use Finagle and have some thoughts on this: @martijnhoekstra @pjfanning @bpholt @dangerousben

Cheers

**Comment by pjfanning** [comment]
After spending nearly a year on Apache Pekko - a fork of Akka (due Akka's license change) - I'm not super anxious to get involved in another large fork.

If a Finagle fork is done, would you expect to soon have to fork other Twitter repos like the twitter/util project?

If X/Twitter were amenable and a small community could be formed, it may be feasible to get the code granted to the ASF and a new Incubator podling formed. I am a member of the Apache Incubator PMC and could help with this transfer. In the absence of X/Twitter granting the code to the ASF, it would be difficult to proceed on the ASF route. It would possibly be easier for someone to just fork FInagle themselves and try making a few releases under their own Maven groupId.

**Comment by felixbr** [comment]
@pjfanning A fork would include twitter/util (as that's the basis for finagle), twitter/finagle, and possibly twitter-server. Of course, I wouldn't expect someone to maintain things they don't use themselves.

I don't think X/Twitter will help in any way, as they haven't even communicated their departure from open-source nor updated the "Status" in readme files. So I expect a fork to be made without their approval or help under a new groupId.

**Comment by dangerousben** [comment]
> I would offer some of my time to keep basic maintenance up (like [Scala versions](https://github.com/twitter/util/issues/274), newer [JDK versions](https://endoflife.date/java), security updates). But I cannot do everything alone as this is a big project and a bus factor of 1 is always bad. My hope would be that devs from other companies which use Finagle will share the responsibility in a collaboration effort.
> 
> I'm tagging some people I've had the pleasure to collaborate with on Finagle and related projects in the past who might also still use Finagle and have some thoughts on this: @martijnhoekstra @pjfanning @bpholt @dangerousben

I'm afraid I'll have to regretfully decline: I'm not currently using scala at work and don't have  the free time for a side project of this nature.

FWIW, although I'm a bit out of the loop, if I were to start a large scala server project today I'd probably start by looking at the [typelevel](https://typelevel.org/projects/) stack.  It's nowhere near as 'batteries included' as finagle, but I believe there are enough third-party additions to bring it up to feature parity.  I couldn't comment on performance, but then who can?

I realise this is no help whatsoever to those maintaining large existing finagle codebases.  Hopefully there are enough people in that position to at least continue basic maintenance.

**Comment by felixbr** [comment]
> I'm afraid I'll have to regretfully decline: I'm not currently using scala at work and don't have the free time for a side project of this nature.

@dangerousben That's perfectly fine. I'm just asking to see who else is using Finagle in a similar situation. If you don't even use Scala currently then this isn't your concern at all and I apologize for tagging you.

Your assertion of the typelevel stack is spot on and we're already using it heavily in many projects (with and without Finagle).

**Comment by bpholt** [comment]
> I would offer some of my time to keep basic maintenance up (like https://github.com/twitter/util/issues/274, newer [JDK versions](https://endoflife.date/java), security updates). But I cannot do everything alone as this is a big project and a bus factor of 1 is always bad. My hope would be that devs from other companies which use Finagle will share the responsibility in a collaboration effort.

I should be able to help with this kind of basic maintenance, at least for a while, but I can't spend a ton of time on it. My company uses Thrift Finagle, but we've been moving to the Typelevel stack for new applications, and this has sparked conversations about accelerating the transition for existing apps as well. 

I'm hoping I'll get some time to work on Scalafix rules or other techniques to make that migration easier. We have the advantage that we never adopted the changes the Scrooge team made to remove the effect parameter in the generated interfaces—in fact, I wrote [Scalafix rules](https://github.com/Dwolla/async-utils/blob/2381051d3380b87694bc168dcf0847ccd235b2b7/scalafix/rules/src/main/scala/com/dwolla/scrooge/scalafix/AdaptHigherKindedThriftCode.scala) to re-introduce the higher kinded type parameter into the generated traits that extend `ThriftService`.

Actually, that reminds me that IMO [twitter/scrooge](https://github.com/twitter/scrooge) should also be included in the list of repos to fork.

**Comment by DieBauer** [comment]
I see more movement here and at twitter/util. @mattdickinson5 do you know what the plans are with open-source Finagle and Util? Can we expect a release, or multiple in the future?

**Comment by martijnhoekstra** [comment]
My company doesn't use any of the twitter stack, my main interest in keeping up updated before was more of a testbed for scala releases themselves, because it exercises a large part of the stack. If you need any concrete help with anything, I'm happy to take a look, but won't have capacity to coordinate anything.

**Comment by csaltos** [comment]
Meanwhile at Twitter, the development department released version "23.11.0" and they are scheduling to launch version "24.2.0" soon.

Certainly Twitter is not releasing versions at a fast an massive rate as before but Finagle is not dead ... Scala and Finagle are the main backbone of Twitter for the backend and they have to keep it alive even at a slow pace.

Additionally Finagle is a mature product, it has a lot of features already a lot of hours of work and it's really good, we use it at my company and is very very fast and performant and more over that you can reason about it ... so to be pragmatic, Finagle at this point does not require a crazy development with releases every week, it's a pity Elon Musk decided to cut resources at Twitter but, again, Finagle has to be kept alive for Twitter be alive.

**Comment by niodice** [comment]
> Meanwhile at Twitter, the development department released version "23.11.0" and they are scheduling to launch version "24.2.0" soon.

@csaltos, is this information stated publicly anywhere?

**Comment by marciopd** [comment]
Hey, guys. Any updates on this?

**Comment by ghost** [comment]
I don't believe that there's any movement at X(.ai) to drive much open-source initiatives. I  believe it to be a large effort to make substantial version updates to things in the Finatra ecosystem, eg Jackson, requiring quite a bit of change inside the codebase for the X(.ai) services.

I'm trying to understand what would be involved in forking the finagle stack - to actually publish new packages - it's got to be something like renaming the projects (due to copyright), and then changing the actual package name from c.t.f. to something else? My main concern in actively maintaining a fork would (initially) be not creating some legal mess from the current "owners".

**Comment by csaltos** [comment]
It's open source, you can fork it anytime, just mention the owners in the new license and you are good to go.

Maintaining a big piece of software like Finagle is very demanding, so it would be great to do it as a team starting with a proposal to the Apache Software foundation to accept the new fork under the wing of Apache Software foundation, like Akka into Pekko by instance.

Then as a team we can all collaborate and make an even better new piece of software based on Finagle using Apache resources and huge help.

**Comment by ghost** [comment]
I get it's open-source - but on various pages there's the "Copyright 2010 Twitter, Inc." statement - so _what_ is copyrighted, and how does one not violate that? ie, if the name is copyrighted.

**Comment by dangerousben** [comment]
Open source software is copyrighted by definition.  It's how the license is
enforced.  So yes, twitter own it, but they have released it under a
license that allows forking it etc.

**Comment by csaltos** [comment]
The industrial property with always belong to Twitter but as long as you mentioned the original authors, everything is fine with the license Twitter decided to use when they share their code base as open source, even in despite of some labeled files, the whole product is open source.
Best regards,
Carlos Saltos
+49 174 839 608 4
***@***.***
Sent from Proton Mail Android
-------- Original Message --------On 4/28/25 10:55, revivebird  wrote:
revivebird left a comment (twitter/finagle#943)
I get it's open-source - but on various pages there's the "Copyright 2010 Twitter, Inc." statement - so what is copyrighted, and how does one not violate that? ie, if the name is copyrighted.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/twitter/finagle/issues/943#issuecomment-2835336818",
"url": "https://github.com/twitter/finagle/issues/943#issuecomment-2835336818",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by rbscgh** [comment]
fwiw there does seem to be some small pushes to util https://github.com/twitter/util/commits/f8735d20dc18acb728fe89bf0373a744e46ae67d/

**Comment by csaltos** [comment]
Thank you very much for revivebird ... it's so valuable and important.

I have one question, the Finagle name can be kept? ... when you fork an open source project you need to give attribution, of course, but also you need to legally change the name to the project to something else ... think of these open source forks:

XFree86 -> X.Org Server
OpenOffice -> LibreOffice
KDE -> TDE
Java -> OpenJDK

Full disclosure, I'm not a lawyer, maybe worth make a professional research beacause all of them changed for legal considerations, trade marks and governance

A new name for the fork could be GoldFeather
